warning: 'register' storage class specifier is deprecated and incompatible with C++1z

```
basic_fun.cpp:4063:11: warning: 'register' storage class specifier is deprecated and incompatible with C++1z [-Wdeprecated-register]
          ELEM_SWAP(array[ll], array[hh])
          ^
/Users/travis/build/slayoo/gdl/src/basic_fun.cpp:4026:26: note: expanded from macro 'ELEM_SWAP'
#define ELEM_SWAP(a,b) { register DFloat t=(a);(a)=(b);(b)=t; }
                         ^
```

--- a/src/basic_fun.cpp
+++ b/src/basic_fun.cpp
@@ -4057,7 +4057,7 @@ namespace lib {
  *  Modified by G. Duvert, 2017, for NaN/INF handling and correction of Nicolas's code
  *  which gave erroneous results when two or more elements were identical. 
  */
-#define ELEM_SWAP(a,b) { register DDouble t=(a);(a)=(b);(b)=t; }
+#define ELEM_SWAP(a,b) { DDouble t=(a);(a)=(b);(b)=t; }
   
   DDouble quick_select_d(DDouble array[], SizeT arraySize, int even) {
 
@@ -4066,7 +4066,7 @@ namespace lib {
     SizeT high, low, middle;
     SizeT median=(arraySize)/2; 
     SizeT ll, hh;
-    register DDouble pivot;
+    DDouble pivot;
     low = 0;
     high = arraySize-1;
     for (;;) {
@@ -4106,7 +4106,7 @@ namespace lib {
   }
 
 #undef ELEM_SWAP
-#define ELEM_SWAP(a,b) { register DFloat t=(a);(a)=(b);(b)=t; }
+#define ELEM_SWAP(a,b) { DFloat t=(a);(a)=(b);(b)=t; }
   
   DFloat quick_select_f(DFloat array[], SizeT arraySize, int even) {
 
@@ -4115,7 +4115,7 @@ namespace lib {
     SizeT high, low, middle;
     SizeT median=(arraySize)/2; 
     SizeT ll, hh;
-    register DFloat pivot;
+    DFloat pivot;
     low = 0;
     high = arraySize-1;
     for (;;) {

--- a/src/envt.cpp
+++ b/src/envt.cpp
@@ -1320,7 +1320,7 @@ bool EnvBaseT::Remove(int* rindx)
 	while( ix >= 0)
 	{
 		inrem++;
-		if(debug)	printf(" env.now.size() = %d  env[%d] = 0x%x ",
+		if(debug)	printf(" env.now.size() = %d  env[%d] = %p ",
 				osz - inrem,
 				ix,static_cast <const void *>(env[ix]) );
 		if ( env[ix] != NULL) GDLDelete( env[ix]);

--- a/src/where.cpp
+++ b/src/where.cpp
@@ -443,7 +443,7 @@ namespace lib {
     SizeT districount[2] = {0, 0};
 
     //distribute accordingly! 
-    register DByte tmp;
+    DByte tmp;
     for (SizeT i = 0; i < nEl; ++i)
     {
       tmp = p0[i];

